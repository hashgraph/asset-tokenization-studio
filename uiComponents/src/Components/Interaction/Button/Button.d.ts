import type { ButtonProps as ChakraButtonProps } from "@chakra-ui/button";
import type { SystemStyleObject, ComponentWithAs } from "@chakra-ui/system";
import type { BaseSingleStyleConfiguration } from "@Theme/Components/BaseSingleStyleConfiguration";
export type ButtonVariantProps = {
    status: ButtonProps["colorScheme"];
};
export interface ButtonThemeConfiguration extends Omit<BaseSingleStyleConfiguration, "variants"> {
    baseStyle: Partial<SystemStyleObject>;
    variants?: {
        [key: string]: ((props: ButtonVariantProps) => Partial<SystemStyleObject>) | Partial<SystemStyleObject>;
    };
}
export interface ButtonProps extends ChakraButtonProps {
    status?: string;
}
export declare const Button: ComponentWithAs<"button", ButtonProps>;
