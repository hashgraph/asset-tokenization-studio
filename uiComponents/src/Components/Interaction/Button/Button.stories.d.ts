import React from "react";
import type { ButtonProps } from "./Button";
declare const meta: import("@storybook/types").ComponentAnnotations<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement>, any, ButtonProps, import("@chakra-ui/system/dist/system.types").As>>;
export default meta;
export declare const Primary: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement>, any, ButtonProps, import("@chakra-ui/system/dist/system.types").As>>;
export declare const PrimaryWithIcon: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement>, any, ButtonProps, import("@chakra-ui/system/dist/system.types").As>>;
export declare const PrimaryWithLoading: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement>, any, ButtonProps, import("@chakra-ui/system/dist/system.types").As>>;
export declare const PrimaryDisabled: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement>, any, ButtonProps, import("@chakra-ui/system/dist/system.types").As>>;
export declare const Secondary: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement>, any, ButtonProps, import("@chakra-ui/system/dist/system.types").As>>;
export declare const SecondaryWithIcon: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement>, any, ButtonProps, import("@chakra-ui/system/dist/system.types").As>>;
export declare const SecondaryWithLoading: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement>, any, ButtonProps, import("@chakra-ui/system/dist/system.types").As>>;
export declare const SecondaryDisabled: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement>, any, ButtonProps, import("@chakra-ui/system/dist/system.types").As>>;
export declare const Tertiary: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement>, any, ButtonProps, import("@chakra-ui/system/dist/system.types").As>>;
export declare const TertiaryWithLoading: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement>, any, ButtonProps, import("@chakra-ui/system/dist/system.types").As>>;
export declare const TertiaryDisabled: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement>, any, ButtonProps, import("@chakra-ui/system/dist/system.types").As>>;
export declare const TertiaryWithIcon: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement>, any, ButtonProps, import("@chakra-ui/system/dist/system.types").As>>;
export declare const Danger: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement>, any, ButtonProps, import("@chakra-ui/system/dist/system.types").As>>;
export declare const DangerWithLoading: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement>, any, ButtonProps, import("@chakra-ui/system/dist/system.types").As>>;
export declare const DangerDisabled: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement>, any, ButtonProps, import("@chakra-ui/system/dist/system.types").As>>;
export declare const DangerWithIcon: import("@storybook/types").AnnotatedStoryFn<import("@storybook/react/dist/types-0a347bb9").R, import("@chakra-ui/system/dist/system.types").MergeWithAs<React.DetailedHTMLProps<React.ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement>, any, ButtonProps, import("@chakra-ui/system/dist/system.types").As>>;
