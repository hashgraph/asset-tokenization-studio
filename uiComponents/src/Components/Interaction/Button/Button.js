var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
import React from "react";
import { Button as ChakraButton } from "@chakra-ui/button";
import { useStyleConfig } from "@chakra-ui/react";
import { forwardRef } from "@chakra-ui/system";
import { Spinner } from "../../Indicators/Spinner/Spinner";
import _merge from "lodash/merge";
export var Button = forwardRef(function (_a, ref) {
    var children = _a.children, sx = _a.sx, status = _a.status, props = __rest(_a, ["children", "sx", "status"]);
    var themeStyles = useStyleConfig("Button", __assign(__assign({}, props), { status: status }));
    var loadingText = typeof children === "string" && !props.loadingText
        ? children
        : props.loadingText;
    var styles = React.useMemo(function () { return _merge(themeStyles, sx); }, [themeStyles, sx]);
    return (React.createElement(ChakraButton, __assign({ ref: ref, loadingText: loadingText, spinner: React.createElement(Spinner, { size: "xxs" }) }, props, { sx: styles }), children));
});
