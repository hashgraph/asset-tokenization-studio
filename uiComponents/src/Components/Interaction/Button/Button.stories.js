var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
import React from "react";
import { Plus, DotsThree } from "@phosphor-icons/react";
import { PhosphorIcon } from "../../Foundations/PhosphorIcon";
import { Button } from "./Button";
import { allButtonVariants, iconLabels, iconList, mappedIcons, allSizes, buttonStatus, } from "@/storiesUtils";
var commonArgs = {
    children: "Button",
    variant: "primary",
    size: "lg"
};
var commonArgsTypes = {
    isDisabled: {
        control: {
            type: "boolean"
        },
        description: "If it is true, the button will be shown as deactivated and cannot be interacted with it. This is a good option to indicate to the user that he cannot perform some action at that time."
    },
    onClick: {
        action: "onClick",
        description: "The function to be called when the button is clicked"
    },
    variant: {
        options: allButtonVariants,
        control: {
            type: "radio"
        },
        description: "Define the visual style of the button. The available options are `Primary`, `Secondary`, `Danger` and `Tertiary`. This is useful when you want to use different buttons styles for different actions."
    },
    size: {
        options: allSizes,
        control: { type: "radio" },
        description: "Define the size of the button. The available options are `md` and `lg`. This can be useful when you want to use different buttons sizes for different devices or for different areas of the user interface where larger or smaller buttons are desired."
    },
    isLoading: {
        control: "boolean",
        description: "Toggle to put the button in loading state"
    }
};
var commonArgsWithIcons = __assign(__assign({}, commonArgs), { leftIcon: React.createElement(PhosphorIcon, { as: Plus }), rightIcon: React.createElement(PhosphorIcon, { as: DotsThree }) });
var commonArgsTypesWithIcons = __assign(__assign({}, commonArgsTypes), { leftIcon: {
        options: iconList,
        control: {
            type: "select",
            labels: iconLabels,
            mapping: mappedIcons,
            required: false
        },
        mapping: mappedIcons,
        description: "Icon to be displayed on the left side of the button"
    }, rightIcon: {
        options: iconList,
        control: {
            type: "select",
            labels: iconLabels,
            mapping: mappedIcons,
            required: false
        },
        mapping: mappedIcons,
        description: "Icon to be displayed on the right side of the button"
    } });
var meta = {
    title: "Design System/Interaction/Button",
    component: Button,
    argTypes: __assign(__assign({}, commonArgsTypes), { status: {
            options: buttonStatus,
            control: { type: "select" }
        } }),
    parameters: {
        design: {
            type: "figma",
            url: "https://www.figma.com/file/YF8AvIq0MclfhGDvKgzypp/NEW---ioBricks-Library?node-id=1503%3A23012&t=uLoNHekrc4UjtgwE-0"
        }
    }
};
export default meta;
var Template = function (args) { return React.createElement(Button, __assign({}, args)); };
// Primary
export var Primary = Template.bind({});
Primary.args = __assign({}, commonArgs);
Primary.argTypes = __assign({}, commonArgsTypes);
export var PrimaryWithIcon = Template.bind({});
PrimaryWithIcon.argTypes = __assign({}, commonArgsTypesWithIcons);
PrimaryWithIcon.args = __assign(__assign({}, commonArgsWithIcons), { variant: "primary" });
export var PrimaryWithLoading = Template.bind({});
PrimaryWithLoading.argTypes = __assign({}, commonArgsTypes);
PrimaryWithLoading.args = __assign(__assign({}, commonArgs), { variant: "primary", isLoading: true, loadingText: "Loading" });
export var PrimaryDisabled = Template.bind({});
PrimaryDisabled.argTypes = __assign({}, commonArgsTypes);
PrimaryDisabled.args = __assign(__assign({}, commonArgs), { variant: "primary", isDisabled: true });
// Secondary
export var Secondary = Template.bind({});
Secondary.args = __assign(__assign({}, commonArgs), { variant: "secondary" });
Secondary.argTypes = __assign({}, commonArgsTypes);
export var SecondaryWithIcon = Template.bind({});
SecondaryWithIcon.argTypes = __assign({}, commonArgsTypesWithIcons);
SecondaryWithIcon.args = __assign(__assign({}, commonArgsWithIcons), { variant: "secondary" });
export var SecondaryWithLoading = Template.bind({});
SecondaryWithLoading.args = __assign(__assign({}, commonArgs), { variant: "secondary", isLoading: true, loadingText: "Loading" });
SecondaryWithLoading.argTypes = __assign({}, commonArgsTypes);
export var SecondaryDisabled = Template.bind({});
SecondaryDisabled.args = __assign(__assign({}, commonArgs), { variant: "secondary", isDisabled: true });
SecondaryDisabled.argTypes = __assign({}, commonArgsTypes);
// Tertiary
export var Tertiary = Template.bind({});
Tertiary.args = __assign(__assign({}, commonArgs), { variant: "tertiary" });
Tertiary.argTypes = __assign({}, commonArgsTypes);
export var TertiaryWithLoading = Template.bind({});
TertiaryWithLoading.args = __assign(__assign({}, commonArgs), { variant: "tertiary", isLoading: true, loadingText: "Loading" });
TertiaryWithLoading.argTypes = __assign({}, commonArgsTypes);
export var TertiaryDisabled = Template.bind({});
TertiaryDisabled.args = __assign(__assign({}, commonArgs), { variant: "tertiary", isDisabled: true });
TertiaryDisabled.argTypes = __assign({}, commonArgsTypes);
export var TertiaryWithIcon = Template.bind({});
TertiaryWithIcon.argTypes = __assign({}, commonArgsTypesWithIcons);
TertiaryWithIcon.args = __assign(__assign({}, commonArgsWithIcons), { variant: "tertiary" });
// Danger
export var Danger = Template.bind({});
Danger.args = __assign(__assign({}, commonArgs), { variant: "primary", status: "error" });
Danger.argTypes = __assign({}, commonArgsTypes);
export var DangerWithLoading = Template.bind({});
DangerWithLoading.args = __assign(__assign({}, commonArgs), { variant: "primary", status: "error", isLoading: true, loadingText: "Loading" });
DangerWithLoading.argTypes = __assign({}, commonArgsTypes);
export var DangerDisabled = Template.bind({});
DangerDisabled.args = __assign(__assign({}, commonArgs), { variant: "primary", status: "error", isDisabled: true });
DangerDisabled.argTypes = __assign({}, commonArgsTypes);
export var DangerWithIcon = Template.bind({});
DangerWithIcon.argTypes = __assign({}, commonArgsTypesWithIcons);
DangerWithIcon.args = __assign(__assign({}, commonArgsWithIcons), { variant: "primary", status: "error" });
