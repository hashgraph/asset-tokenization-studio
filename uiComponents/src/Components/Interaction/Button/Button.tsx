import React from "react";
import type { ButtonProps as ChakraButtonProps } from "@chakra-ui/button";
import type { SystemStyleObject, ComponentWithAs } from "@chakra-ui/system";
import { Button as ChakraButton } from "@chakra-ui/button";
import { useStyleConfig } from "@chakra-ui/react";
import { forwardRef } from "@chakra-ui/system";
import type { BaseSingleStyleConfiguration } from "@Theme/Components/BaseSingleStyleConfiguration";
import { Spinner } from "../../Indicators/Spinner/Spinner";
import _merge from "lodash/merge";

export type ButtonVariantProps = {
  status: ButtonProps["colorScheme"];
};

export interface ButtonThemeConfiguration
  extends Omit<BaseSingleStyleConfiguration, "variants"> {
  baseStyle: Partial<SystemStyleObject>;
  variants?: {
    [key: string]:
      | ((props: ButtonVariantProps) => Partial<SystemStyleObject>)
      | Partial<SystemStyleObject>;
  };
}

export interface ButtonProps extends ChakraButtonProps {
  status?: string;
}

export const Button: ComponentWithAs<"button", ButtonProps> = forwardRef<
  ButtonProps,
  "button"
>(({ children, sx, status, ...props }: ButtonProps, ref) => {
  const themeStyles = useStyleConfig("Button", {
    ...props,
    status,
  });
  const loadingText =
    typeof children === "string" && !props.loadingText
      ? children
      : props.loadingText;

  const styles = React.useMemo(
    () => _merge(themeStyles, sx),
    [themeStyles, sx]
  );

  return (
    <ChakraButton
      ref={ref}
      loadingText={loadingText}
      spinner={<Spinner size="xxs" />}
      {...props}
      sx={styles}
    >
      {children}
    </ChakraButton>
  );
});
