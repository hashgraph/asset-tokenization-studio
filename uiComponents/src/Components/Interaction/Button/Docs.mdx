import {
  Story,
  Meta,
  Title,
  ArgTypes,
  Primary,
  Stories,
  Source,
} from "@storybook/addon-docs";
import { Grid, GridItem, Text, Link, Center, Flex } from "@chakra-ui/react";
import * as ButtonStories from "./Button.stories";

<Meta of={ButtonStories} />

<Title />

<a
  href="https://www.figma.com/file/YF8AvIq0MclfhGDvKgzypp/NEW---ioBricks-Library?node-id=1503%3A23012&t=uLoNHekrc4UjtgwE-0"
  target="\_blank"
>
  View figma design
</a>
<Primary />

## How to set up

Add the following configurations to the `theme.tsx` and `Avatar.ts` files, respectively.

```tsx dark
const theme = {
  components: {
    Button: ConfigButton,
  },
};
```

```tsx dark
const ConfigButton: ButtonThemeConfiguration = {
  baseStyle: {},
  sizes: {},
  variants: {},
};
```

## Props

<ArgTypes of={ButtonStories} />

## Stories

<Stories title="" />
