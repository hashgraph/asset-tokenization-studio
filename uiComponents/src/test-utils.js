import { ChakraProvider, extendTheme } from "@chakra-ui/react";
import { render as testingLibraryRender } from "@testing-library/react";
import React from "react";
import { BasePlatformTheme } from "./Theme/theme";
var DefaultTheme = extendTheme(BasePlatformTheme);
export var render = function (ui, theme) {
    if (theme === void 0) { theme = DefaultTheme; }
    var wrapper = function (_a) {
        var children = _a.children;
        return (React.createElement(ChakraProvider, { theme: theme }, children));
    };
    return testingLibraryRender(ui, { wrapper: wrapper });
};
