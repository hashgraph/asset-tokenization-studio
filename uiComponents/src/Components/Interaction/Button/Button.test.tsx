import { render } from "@/test-utils";
import React from "react";
import type { ButtonProps } from "./Button";
import { Button } from "./Button";

describe(`<Button />`, () => {
  const defaultText = "Content of the button";
  const defaultProps: ButtonProps = {
    children: <>{defaultText}</>,
  };

  const baseSizesList = ["md", "lg"];
  const baseVariantsList = ["primary", "tertiary", "secondary"];
  const statusColors = ["success", "error", "warning", "info"];
  const factoryComponent = (props: ButtonProps = defaultProps) =>
    render(<Button {...props} />);

  test("renders correctly", () => {
    const component = factoryComponent();

    expect(component.asFragment()).toMatchSnapshot();
  });

  test("shows correctly children", () => {
    const label = "Text of the button";
    const component = factoryComponent({ children: <>{label}</> });

    expect(component.getByText(label)).toBeVisible();
  });

  test("pass down props", () => {
    const component = factoryComponent({
      isDisabled: true,
      children: <>{defaultText}</>,
    });

    expect(component.getByText(defaultText)).toBeDisabled();
  });

  baseVariantsList.forEach((variant) => {
    baseSizesList.forEach((size) => {
      statusColors.forEach((status) => {
        test(`renders correctly for size ${size} & variant ${variant}`, () => {
          const component = factoryComponent({
            ...defaultProps,
            size,
            variant,
            status,
          });
          expect(component.asFragment()).toMatchSnapshot(
            `Button-${variant}-${size}-${status}`
          );
        });
      });
    });
  });
});
