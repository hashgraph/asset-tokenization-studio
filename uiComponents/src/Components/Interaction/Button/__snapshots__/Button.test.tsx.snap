// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> renders correctly 1`] = `
<DocumentFragment>
  <button
    class="chakra-button css-1c8yzl7"
    type="button"
  >
    Content of the button
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Button /> renders correctly for size lg & variant primary: Button-primary-lg-error 1`] = `
<DocumentFragment>
  <button
    class="chakra-button css-1p4yh7l"
    type="button"
  >
    Content of the button
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Button /> renders correctly for size lg & variant primary: Button-primary-lg-info 1`] = `
<DocumentFragment>
  <button
    class="chakra-button css-493doh"
    type="button"
  >
    Content of the button
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Button /> renders correctly for size lg & variant primary: Button-primary-lg-success 1`] = `
<DocumentFragment>
  <button
    class="chakra-button css-17bas7"
    type="button"
  >
    Content of the button
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Button /> renders correctly for size lg & variant primary: Button-primary-lg-warning 1`] = `
<DocumentFragment>
  <button
    class="chakra-button css-1bfguqv"
    type="button"
  >
    Content of the button
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Button /> renders correctly for size lg & variant secondary: Button-secondary-lg-error 1`] = `
<DocumentFragment>
  <button
    class="chakra-button css-c6c3sq"
    type="button"
  >
    Content of the button
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Button /> renders correctly for size lg & variant secondary: Button-secondary-lg-info 1`] = `
<DocumentFragment>
  <button
    class="chakra-button css-wcx3ir"
    type="button"
  >
    Content of the button
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Button /> renders correctly for size lg & variant secondary: Button-secondary-lg-success 1`] = `
<DocumentFragment>
  <button
    class="chakra-button css-yczomy"
    type="button"
  >
    Content of the button
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Button /> renders correctly for size lg & variant secondary: Button-secondary-lg-warning 1`] = `
<DocumentFragment>
  <button
    class="chakra-button css-9r1lkf"
    type="button"
  >
    Content of the button
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Button /> renders correctly for size lg & variant tertiary: Button-tertiary-lg-error 1`] = `
<DocumentFragment>
  <button
    class="chakra-button css-1ule0dh"
    type="button"
  >
    Content of the button
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Button /> renders correctly for size lg & variant tertiary: Button-tertiary-lg-info 1`] = `
<DocumentFragment>
  <button
    class="chakra-button css-1524wrs"
    type="button"
  >
    Content of the button
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Button /> renders correctly for size lg & variant tertiary: Button-tertiary-lg-success 1`] = `
<DocumentFragment>
  <button
    class="chakra-button css-j0go3y"
    type="button"
  >
    Content of the button
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Button /> renders correctly for size lg & variant tertiary: Button-tertiary-lg-warning 1`] = `
<DocumentFragment>
  <button
    class="chakra-button css-nt47wb"
    type="button"
  >
    Content of the button
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Button /> renders correctly for size md & variant primary: Button-primary-md-error 1`] = `
<DocumentFragment>
  <button
    class="chakra-button css-dwbvxv"
    type="button"
  >
    Content of the button
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Button /> renders correctly for size md & variant primary: Button-primary-md-info 1`] = `
<DocumentFragment>
  <button
    class="chakra-button css-2h6xjp"
    type="button"
  >
    Content of the button
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Button /> renders correctly for size md & variant primary: Button-primary-md-success 1`] = `
<DocumentFragment>
  <button
    class="chakra-button css-111wghi"
    type="button"
  >
    Content of the button
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Button /> renders correctly for size md & variant primary: Button-primary-md-warning 1`] = `
<DocumentFragment>
  <button
    class="chakra-button css-9bpvsg"
    type="button"
  >
    Content of the button
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Button /> renders correctly for size md & variant secondary: Button-secondary-md-error 1`] = `
<DocumentFragment>
  <button
    class="chakra-button css-jrquix"
    type="button"
  >
    Content of the button
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Button /> renders correctly for size md & variant secondary: Button-secondary-md-info 1`] = `
<DocumentFragment>
  <button
    class="chakra-button css-og06iw"
    type="button"
  >
    Content of the button
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Button /> renders correctly for size md & variant secondary: Button-secondary-md-success 1`] = `
<DocumentFragment>
  <button
    class="chakra-button css-1d200yb"
    type="button"
  >
    Content of the button
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Button /> renders correctly for size md & variant secondary: Button-secondary-md-warning 1`] = `
<DocumentFragment>
  <button
    class="chakra-button css-1x1zowq"
    type="button"
  >
    Content of the button
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Button /> renders correctly for size md & variant tertiary: Button-tertiary-md-error 1`] = `
<DocumentFragment>
  <button
    class="chakra-button css-o2nbyp"
    type="button"
  >
    Content of the button
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Button /> renders correctly for size md & variant tertiary: Button-tertiary-md-info 1`] = `
<DocumentFragment>
  <button
    class="chakra-button css-uo9j4h"
    type="button"
  >
    Content of the button
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Button /> renders correctly for size md & variant tertiary: Button-tertiary-md-success 1`] = `
<DocumentFragment>
  <button
    class="chakra-button css-5lfg67"
    type="button"
  >
    Content of the button
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`<Button /> renders correctly for size md & variant tertiary: Button-tertiary-md-warning 1`] = `
<DocumentFragment>
  <button
    class="chakra-button css-1s06x8d"
    type="button"
  >
    Content of the button
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
